x/transferinject/ibc_middleware_test.go

```diff
@@ -0,0 +1,461 @@
+package transferinject_test
+
+import (
+	"fmt"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	transfertypes "github.com/cosmos/ibc-go/v6/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v6/modules/core/04-channel/types"
+	porttypes "github.com/cosmos/ibc-go/v6/modules/core/05-port/types"
+	"github.com/stretchr/testify/require"
+
+	rollapptypes "github.com/dymensionxyz/dymension/v3/x/rollapp/types"
+	"github.com/dymensionxyz/dymension/v3/x/transferinject"
+	"github.com/dymensionxyz/dymension/v3/x/transferinject/types"
+)
+
+func TestIBCMiddleware_SendPacket(t *testing.T) {
```

Comment:
need to use errorsmod.IsOf or errors.Is instead of string comparison for all these errors

Reply:
Some of the subtest names could be better 
This one is OK: `name: "return early: rollapp already has token metadata",`
Because it tells me what is being tested (the fact that we catch it if it already has data)

But the rest of them dont really tell me what's being tested, they just say what happens or what is returned